pandas/core/arrays/timedeltas.py

```diff
@@ -137,25 +139,67 @@ def dtype(self):
     # Constructors
     _attributes = ["freq"]
 
-    @classmethod
-    def _simple_new(cls, values, freq=None, dtype=_TD_DTYPE):
-        # `dtype` is passed by _shallow_copy in corner cases, should always
-        #  be timedelta64[ns] if present
-        assert dtype == _TD_DTYPE
-        assert isinstance(values, np.ndarray), type(values)
+    def __init__(self, values, dtype=_TD_DTYPE, freq=None, copy=False):
```

Comment:
This seems like a lot of new code.  There's no viable way to push it off into sequence_to_dt64ns or _from_sequence?

Reply:
That's what I did earlier, though I struggled with errors from a frequency validation check (that may have been datetime array, but they're basically the same).

Reply:
>  though I struggled with errors from a frequency validation check 

Was it a pickle-based test failure?  There were a bunch of these during the last round of DTA/DTI constructor stuff I did, eventually got it trimmed down to one, and that last one turns out to be a bug in `_generate_range_overflow_safe` that is fixed by #24255.

Regardless, the non-validating constructors should be private.  (OK, I'll stop beating this dead horse for now)

Reply:
Since #24255 has been merged, the pickle-based error should be resolved.  Does this improve share-ability?